[Rui Cong] iP

thamruicong opened the pull request:
# aRC
> "When you have a dream, you've got to grab it and never let go." – Carol Burnett [(source)](https://parade.com/973277/jessicasager/inspirational-quotes/)
---
[aRC](https://github.com/thamruicong/ip) is ...
- a checklist made by *yours truly*
- easy to pick up, hard to master
- **addictive**

This checklist supports **THREE** different tasks
1. Todo
2. Deadline
3. Event

Some of the features includes:
- [x] Creating tasks 📓 
- [x] Marking and unmarking tasks 💯  
- [x] Deleting tasks ☠️ 
- [x] Finding tasks 🔍 
- [ ] ~~Doing your tasks for you~~ 🤬 
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

thamruicong committed:
Level-1 done

thamruicong committed:
Level-2 done

thamruicong committed:
Level-3 done

thamruicong committed:
Level-4 done

thamruicong committed:
Level-5 done

thamruicong committed:
Level-6 done

thamruicong committed:
Minor style

thamruicong committed:
customized aRC and added command syntax on exception

thamruicong committed:
A-TestUiTesting done

thamruicong committed:
Changed name of ToDo.java to Todo.java

thamruicong committed:
Added switch statement and javaDocs

thamruicong committed:
changed print() to toString()

thamruicong committed:
Refactor Tasklist.java

thamruicong committed:
Level-7

thamruicong committed:
Handle create dir

thamruicong committed:
Merge branch 'branch-Level-7'

thamruicong committed:
Level-8 done

thamruicong committed:
Merge branch 'branch-Level-8'

thamruicong committed:
Rename DukeControl to Parser

thamruicong committed:
A-MoreOOP done

thamruicong committed:
A-Packages done

thamruicong committed:
A-Gradle done

thamruicong committed:
A-JUnit done

thamruicong committed:
A-JavaDoc done

thamruicong committed:
A-CodingStandard done

thamruicong committed:
Level-9 done

thamruicong committed:
Merge branch 'branch-A-JavaDoc'

thamruicong committed:
Merge branch 'branch-A-CodingStandard'

thamruicong committed:
Merge branch 'branch-Level-9'

thamruicong committed:
Add JavaDocs for parseFind

thamruicong committed:
Validate index and dateTime

thamruicong committed:
Change code to resolve checkstyle violations

Samsation commented on src/main/java/aRC/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+     * Constructor for TaskList
+     * @param arrayList An ArrayList of Tasks
+     */
+    public TaskList(ArrayList<Task> arrayList) {
```
Perhaps this can be named 'tasks' for extra clarity

Samsation commented on src/main/java/aRC/UI.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Prints hello message to the screen
+     */
+    public void hello() {
```
Maybe you can consider naming this displayHelloMessage() instead for added clarity 😄

Samsation commented on src/main/java/aRC/UI.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Prints goodbye message to the screen
+     */
+    public void bye() {
```
Similar point as above

Samsation commented on src/main/java/aRC/Parser.java:
```diff
@@ -0,0 +1,243 @@
+     * @return The integer index if it is valid
+     * @throws InvalidArgumentException If the index String is invalid
+     */
+    public int validateIndex(String index) throws InvalidArgumentException {
```
I really liked how you separated the checks into a separate. Makes the entire implementation neater and more readable 👍

Samsation commented on src/main/java/aRC/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+        }
+
+        for (int i = 0; i < this.arrayList.size(); i++) {
+            if (this.arrayList.get(i).title.toLowerCase().contains(keyword.toLowerCase())) {
```
Perhaps you can declare this predicate as a boolean variable with a name as it is not immediately obvious what it does

Samsation reviewed (commented):
Overall, I found your code to be really easy to follow and understand. Great job!

thamruicong committed:
Add JavaFX GUI

thamruicong committed:
Close on bye command

thamruicong committed:
Merge branch 'branch-Level-10'

thamruicong committed:
Add delay before bye command

thamruicong committed:
Add assertions to Parser.java

thamruicong committed:
Improve Code Quality

* Change taskList to tasks
* Remove arrowhead style code from Parser.java
* Add containsRegardlessOfCase() method to TaskList.java
* Refactor load() method in Storage.java

thamruicong committed:
Merge pull request #1 from thamruicong/branch-A-Assertions

Add assertions to Parser.java

thamruicong committed:
Merge branch 'master'

thamruicong committed:
Merge pull request #2 from thamruicong/branch-A-CodeQuality

Improve code quality

thamruicong committed:
Add CI workflow

thamruicong committed:
Checkstyle changes

thamruicong committed:
Change textbox to stretch on long text

thamruicong committed:
Refactor ParserTest.java testcase

thamruicong committed:
Add note list,add,delete functionality

thamruicong committed:
Add load and save for Notes

thamruicong committed:
Merge branch 'branch-D-Notes'

thamruicong committed:
Minor checkstyle changes

thamruicong committed:
Merge branch 'branch-D-Notes'

thamruicong committed:
Change Duke and User images

- Clip images to circle

thamruicong committed:
Add Background image

thamruicong committed:
Add StackPane

thamruicong committed:
Improve GUI

thamruicong committed:
Add User Guide

Modify docs/README.md User Guide. Add UI screenshots of aRC application.
Add application title "aRC Application".

thamruicong committed:
Fix screaming snake case checkstyle issue

thamruicong committed:
Change build.gradle mainClassName

thamruicong committed:
Update README.md
